pydivide/utilities.py

```diff
@@ -494,35 +494,38 @@ def get_latest_files_from_date_range(date1, date2):
         day = str('%02d' % current_date.day)
         full_path = os.path.join(maven_data_dir, year, month)
         if os.path.exists(full_path):
-            versions = []
-            revisions = []
-            crustal = []
+            version = 0
+            revision = 0
+            c_version = 0
+            c_revision = 0
+
+            # Grab only the latest versions/revisions of regular and crustal insitu files
             i = 0
             for f in os.listdir(full_path):
-                if kp_regex.match(f).group('day') == day:
+                if kp_regex.match(f).group('day') == day and not kp_regex.match(f).group('description'):
                     v = kp_regex.match(f).group('version')
-                    if int(v) > 0:
+                    if int(v) > int(version):
                         version = v
-                        versions.append(version)
                     r = kp_regex.match(f).group('revision')
-                    if int(r) > 0:
+                    if int(r) > int(revision):
```

Comment:
This will find only the highest version and highest revision, even if those two numbers come from two different files.  So if the user has a v15_r01 file and a v14_r08 file, this will ultimately wind up returning a file that is "v15_r08"